compilers/concrete-compiler/compiler/lib/Dialect/FHE/Analysis/MANP.cpp

```diff
@@ -1129,19 +558,9 @@ static llvm::APInt getSqMANP(mlir::tensor::ExtractOp op,
   return eNorm;
 }
 
-static llvm::APInt getSqMANP(FHELinalg::FromElementOp op,
-                             llvm::ArrayRef<const MANPLattice *> operandMANPs) {
-
-  auto manp = operandMANPs[0]->getValue().getMANP();
-  if (manp.has_value()) {
-    return manp.value();
-  }
-
-  return llvm::APInt{1, 1, false};
-}
-
-static llvm::APInt getSqMANP(mlir::tensor::FromElementsOp op,
-                             llvm::ArrayRef<const MANPLattice *> operandMANPs) {
+static std::optional<llvm::APInt>
+getSqMANP(mlir::tensor::FromElementsOp op,
```

Comment:
Ok I see you dan't use the interfaces mechanism for ops that we don't define, maybe there are a way to attach interface to ops like in the bufferization mechanism (maybe @andidr you know how to achieve that).

Reply:
Its is possible to do that with external models. The file [`compilers/concrete-compiler/compiler/lib/Dialect/Concrete/Transforms/BufferizableOpInterfaceImpl.cpp`](https://github.com/zama-ai/concrete/blob/e46cf4d85bcae75364d033951149d4ac871e44a4/compilers/concrete-compiler/compiler/lib/Dialect/Concrete/Transforms/BufferizableOpInterfaceImpl.cpp) shows this for `BufferizableOpInterface`. You just need to ensure that  the code attaching the interface is executed at the beginning (e.g., by adding something to `CompilationContext::getMLIRContext()` in  [`compilers/concrete-compiler/compiler/lib/Support/CompilerEngine.cpp`](https://github.com/zama-ai/concrete/blob/e46cf4d85bcae75364d033951149d4ac871e44a4/compilers/concrete-compiler/compiler/lib/Support/CompilerEngine.cpp)).